### DISCUSSION TITLE
How can I change the font size of a button and a label separately ?

### DISCUSSION THREAD

**Discussion by Slayne97** [israel-dryer/ttkbootstrap#371] [Q&A]
I am making a simple GUI using ttkbootstrap, but I can't seem to find a way to change de font size of a label and a button separately. 
So far I've managed to find this: 
`
default_font = font.nametofont('TkDefaultFont')
`
`
default_font['size'] = 24
`
But that changes all font sizes. Thank you in advance.

**Comment by israel-dryer** [comment]
The font is a style level configuration, so you need to set the font of the underlying widget style.

<img width="113" alt="image" src="https://user-images.githubusercontent.com/42584742/200580822-609b6d56-61a0-4666-9193-4ce858c5ff83.png">


```python
import ttkbootstrap as ttk

app = ttk.Window()
style = ttk.Style()

# add a few test buttons
ttk.Button(text='Push 1').pack(padx=20, pady=20)
ttk.Button(text='Push 2', bootstyle='info').pack(padx=20, pady=20)

# updating the style here...
style.configure('info.TButton', font='-size 25')

app.mainloop()
```

**Reply by aDramaQueen** [reply]
Hi,
does this only work with "normal" buttons? As you can see in the image, just the first one has the defined properties.

I also tried all combinations from your [**Documentation**](https://ttkbootstrap.readthedocs.io/en/latest/gettingstarted/tutorial/#keyword-usage). No luck so far...

```python
def do_somthing():
    pass

def tkinter_example():

    root = Window()
    style = ttkbootstrap.Style('literia')

    frame = ttkb.Frame(root)

    btn1 = ttkb.Button(frame, text='INFO', style='info', command=do_something)
    btn1.pack()
    style.configure('info.TButton', font=('arial', 18, 'bold'))

    btn2 = ttkb.Button(frame, text='INFO-OUTLINE', style='info-outline', command=do_something)
    btn2.pack()
    style.configure('info-outline.TButton', font=('arial', 18, 'bold'))

    btn3 = ttkb.Button(frame, text='INFO-LINK', style='info-link', command=do_something)
    btn3.pack()
    style.configure('info-link.TButton', font=('arial', 18, 'bold'))

    frame.pack()
    root.mainloop()


if __name__ == '__main__':
    tkinter_example()
```

Result:

![grafik](https://user-images.githubusercontent.com/52710085/208754422-e62e84cb-07d7-4ce1-a97a-649c30afdd27.png)

**Reply by peetaCodes** [reply]
Thank you!
I'm wondering though, can't you make it so that those properties are tied to the single widget?
In tkinter I can change the font of a single button, while here this isn't possible, and I can just change the global theme for every one of the buttons of that style to have that font.

I just wonder why you've chosen to set the font of a button as a style-level configuration and not as a widget-level configuration as it is in tkinter. I mean, ttkbootstrap is just a wrapper for tkinter, so I think it would make more sense to preserve those kinds of features

**Reply by israel-dryer** [reply]
The Tkinter widgets are not tied to the styling system unlike the ttk widgets. This is why you can make widget level adjustments... They are not themed.  This is built into the underlying tcltk architecture 

However, in the upcoming v2, widget level styles are created based on the parameters you supply in the constructor, so in v2 you will technically be able to make widget level font changes as it will automatically generate a new style for that widget.

You can change the font now,  you just have to do it through the theming system.

**Reply by peetaCodes** [reply]
Thank you!
Sorry for replying late, I just read your message.
I've learnt that I'm actually able of creating new styles that are based on older ones, so I've used that trick to manage to have multiple buttons of the same boot style to have different fonts.

I'm still excited for the upcoming of ver2, though!

**Reply by israel-dryer** [reply]
@peetaCodes @TheDemonGuard @Slayne97, I've created a discussion about the upcoming version 2 release.  I'll be posting previews and other things up until the release to give everyone an idea of what is coming.  https://github.com/israel-dryer/ttkbootstrap/discussions/733

**Comment by israel-dryer** [comment]
The format of the underlying ttk style is `info.Outline.TButton` and `info.Link.TButton`

What happens is that the keywords are parsed and converted to a style that is recognized by the underlying tcl/tk interpreter. If you ever need to see what the actual ttk style name is (for subclassing for example), you can use `.configure('style')` to find it on any ttk widget... for example `btn1.configure('style')`

**Comment by TheDemonGuard** [comment]
@Slayne97 
For reference, similar topic was discussed #322 as well.

> I am making a simple GUI using ttkbootstrap, but I can't seem to find a way to change de font size of a label and a button separately. So far I've managed to find this: `default_font = font.nametofont('TkDefaultFont')` `default_font['size'] = 24` But that changes all font sizes. Thank you in advance.
